Infrastructure: update tr() annotation style

tachyonicbytes opened the pull request:
#### Brief overview of PR changes/additions
Update the tr() annotation style, as discussed in #5522 
#### Motivation for adding to Mudlet
Makes the code cleaner and also keeps Mudlet in tune with the latest standards.
#### Other info (issues closed, discussion etc)
/claim #5522

tachyonicbytes committed:
#5522 Update tr() annotation style

SlySven commented on src/TMainConsole.cpp:
```diff
@@ -176,7 +176,8 @@ void TMainConsole::toggleLogging(bool isMessageEnabled)
             if (mpHost->mLogFileName.isEmpty()) {
                 // If no log name is set, use the default placeholder
-                logFileName = tr("logfile", "Must be a valid default filename for a log-file and is used if the user does not enter any other value (Ensure all instances have the same translation {2 of 2}).");
+                //: Must be a valid default filename for a log-file and is used if the user does not enter any other value (Ensure all instances have the same translation {2 of 2}).
```
Actually when I indexed these duplicated translatable `QString`s by adding a `{X of Y}` suffix in order to try and ensure that if one copy got changed other coders would realise that there were other `Y-1` cases to change I messed up and instead should have just said something like `{one of Y copies}` for ALL of them. The latter would mean that the Qt utilities would see an **identical** source in those Y cases whereas the former breaks that because `X` is different for each one and thus are NOT seen as identical texts... :man_facepalming:

tachyonicbytes replied:
Ah, I see it now. What's the solution here? Should I just write `{one of 2 copies}`, for this example?

tachyonicbytes replied:
That was the only one that I could find. I added them as `{one of two copies}`, as it seemed strange to me to add two numbers, one in arabic numerals and the other in latin script.

SlySven commented on src/TMap.cpp:
```diff
@@ -2563,7 +2564,8 @@ bool TMap::readXmlMapFile(QFile& file, QString* errMsg)
         isLocalImport = true;
         mpProgressDialog = new QProgressDialog(tr("Importing XML map file for use in %1...").arg(mProfileName), QString(), 0, 0);
-        mpProgressDialog->setWindowTitle(tr("Map import", "This is a title of a progress dialog."));
+        //: This is a title of a progress dialog.
```
Actually, given the other two texts it would probably be better to revise the comment:
```suggestion
        //: This is a title of a progress window.
```

SlySven commented on src/ctelnet.cpp:
```diff
@@ -480,11 +480,12 @@ void cTelnet::slot_socketDisconnected()
-                                            "This is the format to be used to show the profile connection time, it follows "
-                                            "the rules of the \"QDateTime::toString(...)\" function and may need "
-                                            "modification for some locales, e.g. France, Spain.")));
+                                /*:
+                                This is the format to be used to show the profile connection time, it follows
+                                the rules of the \"QDateTime::toString(...)\" function and may need
+                                modification for some locales, e.g. France, Spain.
+                                */
+                               .toString(tr("hh:mm:ss.zzz")));
```
Since this is no longer a string argument to a function the double-quotes do not need escaping now:
```suggestion
                                /*:
                                This is the format to be used to show the profile connection time, it follows
                                the rules of the "QDateTime::toString(...)" function and may need
                                modification for some locales, e.g. France, Spain.
                                */
                               .toString(tr("hh:mm:ss.zzz")));
```

tachyonicbytes replied:
Yeah, I got this right in many places, but I missed this somehow.

SlySven commented on src/ctelnet.cpp:
```diff
@@ -2374,7 +2379,8 @@ void cTelnet::postMessage(QString msg)
             QString firstLineTail = body.at(0).mid(prefixLength);
             body.removeFirst();
-            if (prefix.contains(tr("ERROR", "Keep the capisalisation, the translated text at 7 letters max so it aligns nicely")) || prefix.contains(QLatin1String("ERROR"))) {
+            //: Keep the capisalisation, the translated text at 7 letters max so it aligns nicely
```
Given that you are now editing this text the Spelling check bot would start moaning about this (repeated) "smelling error" someone (possibly me) made - if it was still working!
```suggestion
            //: Keep the capitalisation, the translated text at 7 letters max so it aligns nicely
```

tachyonicbytes replied:
Didn't think of this, thanks!

SlySven commented on src/dlgPackageExporter.cpp:
```diff
@@ -1462,14 +1461,11 @@ void dlgPackageExporter::slot_recountItems(QTreeWidgetItem *item)
-                                             // Intentional comment to separate arguments
-                                             "This is the text shown at the top of a groupbox when there is %n (one or more) items to export in the Package exporter dialogue; the initial (and when there is no items selected) is a separate text.",
-                                             itemsToExport));
+                //: This is the text shown at the top of a groupbox when there is %n (one or more) items to export in the Package exporter dialogue; the initial (and when there is no items selected) is a separate text.
```
Do we know that this form works when there is an empty string `""` as the second argument to the `QObject::tr(...)` call when there is a third (the quantity used to drive the handling of the plural forms for "numerus" translations), given that the default argument is actually a `nullptr`? I.e. will it override the `""` and this `//:` prefix form comment be used or will the `""` win out?

tachyonicbytes replied:
Ahh, missed that about the `nullptr`, seems much more logical as a default argument. I relied on the initial discussion too much for the code's benefit. I will update them all, thanks.

The other problem is that I don't know where to look in Qt Designer to find this exact string, in order to test that the changes propagate.

tachyonicbytes replied:
Added `nullptr` for all of them.

SlySven replied:
> The other problem is that I don't know where to look in Qt Designer to find this exact string, in order to test that the changes propagate.

You will actually want to run `lupdate` on the source code - go to the base of your local Mudlet (working) repository and do (assuming you are on a Unix-ish OS):
```sh
lupdate -pro ./src/mudlet.pro -ts ./translations/translated/mudlet.ts
```
note that this will probably change that `mudlet.ts` file but it is not a file we track in the main git repository so that is probably fine...

Then you will want to view the results of the changes you have made in the source code and to do that you will want to fire up the `linguist` tool which is what someone doing the translations themselves would probably use but we don't because we do (nearly all of it) online, collaboratively at **[CrowdIn](https://crowdin.com/project/mudlet)** which our CI/CB scripts do the `lupdate` and injects that file into their system which results in a batch of `mudlet_xx_YY.ts` one for each locale. 
```sh
linguist ./translation/translated/mudlet.ts
```

Anyhow, you will hopefully find the texts that you have modified and there be able to see if the "Developer comments" have made it through. There are a few of the new style annotations already in the file so you can see this in the following screen shot:
![image](https://github.com/Mudlet/Mudlet/assets/6163092/54ed9aa9-c882-4e42-b104-b54f2ffce63d)

You may find that as well as `lupdate` and `linguist` you have IIRC some similar named tools with "pro" at the end, which Qt suggest are the "modern" replacements but you don't need to use them as the "originals" do the job fine...

SlySven commented on src/dlgProfilePreferences.cpp:
```diff
@@ -3806,15 +3810,22 @@ void dlgProfilePreferences::generateDiscordTooltips()
-                                  .arg(tr("Time", "Discord Rich Presence time until or time elapsed"));
+                                       //: Discord Rich Presence party size
+                                       tr("Party size"),
+                                       //: , Discord Rich Presence maximum party size
```
```suggestion
                                       //: Discord Rich Presence maximum party size
```

SlySven commented on src/dlgRoomExits.cpp:
```diff
@@ -1047,62 +1047,50 @@ QAction* dlgRoomExits::getActionOnExit(QLineEdit* pExitLineEdit) const
-                                       tr("<b>Room is locked</b>, it will not be used for speed-walks for any exit that leads to it.",
-                                          // Intentional comment to separate arguments
-                                          "Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock (\"No route\") setting of any exit that goes to it."));
+                                       //: Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock (\"No route\") setting of any exit that goes to it.
```
```suggestion
                                       //: Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock ("No route") setting of any exit that goes to it.
```

SlySven commented on src/dlgRoomExits.cpp:
```diff
@@ -1047,62 +1047,50 @@ QAction* dlgRoomExits::getActionOnExit(QLineEdit* pExitLineEdit) const
-                                       tr("<b>Room is locked</b>, it will not be used for speed-walks for any exit that leads to it.",
-                                          // Intentional comment to separate arguments
-                                          "Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock (\"No route\") setting of any exit that goes to it."));
+                                       //: Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock (\"No route\") setting of any exit that goes to it.
```
```suggestion
                                       //: Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock ("No route") setting of any exit that goes to it.
```

SlySven commented on src/dlgRoomExits.cpp:
```diff
@@ -1047,62 +1047,50 @@ QAction* dlgRoomExits::getActionOnExit(QLineEdit* pExitLineEdit) const
-                                   tr("<b>Room is locked</b>, it will not be used for speed-walks for any exit that leads to it.",
-                                      // Intentional comment to separate arguments
-                                      "Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock (\"No route\") setting of any exit that goes to it."));
+                                   //: Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock (\"No route\") setting of any exit that goes to it.
```
```suggestion
                                   //: Bold HTML tags are used to emphasis that destination room locked status overrides any weight or lock ("No route") setting of any exit that goes to it.
```

SlySven commented on src/dlgRoomExits.cpp:
```diff
@@ -1519,9 +1507,8 @@ void dlgRoomExits::init()
-        roomID->setToolTip(utils::richText(tr("This is the Room ID number for this room; this <b>room is locked</b> so it will not be used for speed-walks at all.",
-                                              // Intentional comment to separate arguments
-                                              "This text is a revision to the default tooltip text set for this widget in the 'room_exits.ui' file. Bold HTML tags are used to emphasis that this room's locked status overrides any weight or lock (\"No route\") setting of any exit that comes to it.")));
+        //: This text is a revision to the default tooltip text set for this widget in the 'room_exits.ui' file. Bold HTML tags are used to emphasis that this room's locked status overrides any weight or lock (\"No route\") setting of any exit that comes to it.
```
```suggestion
        //: This text is a revision to the default tooltip text set for this widget in the 'room_exits.ui' file. Bold HTML tags are used to emphasis that this room's locked status overrides any weight or lock ("No route") setting of any exit that comes to it.
```

tachyonicbytes committed:
Format fixing

Apply the easier format fixing from the GitHub feedback.

Co-authored-by: Stephen Lyons <slysven@virginmedia.com>

tachyonicbytes committed:
Update tr() invokations with three parameters.

Update all the tr() calls that have three parameters. The second
parameter is replaced with `nullptr`, the actual default parameter,
instead of the `""` that is currently present.

tachyonicbytes committed:
Make identical strings identical again.

Some translator comments strings were meant to be identical, but had a
suffix added that made them different. Make them identical again, so
that they appear identical in Qt Designer.

SlySven commented:
You will also want to check that those multi-line `/*: ..... */` comments do not get all that white-space at the beginning of each line (as an indent) into them when seen in `linguist` - in other places I have put *non-* "translator" comments in with intermediate lines all start with a `* ` - I dunno whether that would work either when moving multi-line texts out of the `tr(...)`...

:man_shrugging:

tachyonicbytes commented:
It seems that the comments are recorded without the leading whitespaces.
![image](https://github.com/Mudlet/Mudlet/assets/97369296/fec01558-6a80-4354-bd2f-474faccfb749)
![image](https://github.com/Mudlet/Mudlet/assets/97369296/b7ddc3eb-1819-4b35-af53-661b79d5cb26)
The screenshots show that comment with multiline comment syntax. So the comments are updated, and the multiline syntax works as well. Please let me know if there is anything else to check.

SlySven commented:
> It seems that the comments are recorded without the leading whitespaces. ![image](https://user-images.githubusercontent.com/97369296/239955394-fec01558-6a80-4354-bd2f-474faccfb749.png) ![image](https://user-images.githubusercontent.com/97369296/239955521-b7ddc3eb-1819-4b35-af53-661b79d5cb26.png) The screenshots show that comment with multiline comment syntax. So the comments are updated, and the multiline syntax works as well. Please let me know if there is anything else to check.

Ah, if you look really closely at the first line you can see that all the whitespace (the end-of-line AND all the spaces of the indent) that follows the hyphen in "command-line"  is converted/collapsed to a ***single*** space in the display in `linguist` - which is not exactly what it would be hoped it would be converted to (nothing at all)! :cry: OTOH for ***most* lines** that will still work.

Thanks for testing this though as it gives us valuable information. I also guess then that a leading `* ` on each line isn't going to fair well in this situation either.

tachyonicbytes commented:
I think the usual C backslash behavior of deleting the new line and making one logical line from two physical lines applies here. I will try to check with
```
command-\
line
```
which should end up as `command-line`.

vadi2 reviewed (approved):
Thank you! This is simpler to look at for sure.
